### PR TITLE
Feature/enemy

### PR DESCRIPTION
**Pull request by JoepEilander**
first two enemy spritesheets and flat sprites

### CODE REVIEW

File: Assets/Art/2D/Sprites/Enemies/Spritesheets/AC_Enemy_Blue_512x512_0.controller
Comment (lucasvdiepen) [inline Assets/Art/2D/Sprites/Enemies/Spritesheets/AC_Enemy_Blue_512x512_0.controller:1]: Remove 512x512 and underscores in file name

File: Assets/Art/2D/Sprites/Enemies/Spritesheets/AC_Enemy_Green_512x512_0.controller
Comment (lucasvdiepen) [inline Assets/Art/2D/Sprites/Enemies/Spritesheets/AC_Enemy_Green_512x512_0.controller:1]: Remove 512x512 and underscores in file name

File: Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        [SerializeField]
+        private SpriteRenderer _spriteRenderer;
+
+        private float _lerpAlpha = 1;
+
+        private void Start()
+        {
+            if(_spriteRenderer == null)
+                _spriteRenderer = gameObject.GetComponent<SpriteRenderer>();
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs:22]: GetComponent in `Awake`

File: Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+using ProefExamen.Framework.Gameplay.Values;
+using System.Collections;
+using UnityEngine;
+
+namespace ProefExamen.Framework.Gameplay.LaneSystem
+{
+    public class DeadNote : MonoBehaviour
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs:12]: Missing summery also needs [RequireComponent(typeof(SpriteRenderer))]

File: Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+        private float _lerpAlpha = 1;
+
+        private void Start()
+        {
+            if(_spriteRenderer == null)
+                _spriteRenderer = gameObject.GetComponent<SpriteRenderer>();
+        }
+
+        public void SetDeadNoteValues(Sprite sprite, Transform newTransform)
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs:30]: Missing summery

File: Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
+using ProefExamen.Framework.Gameplay.Values;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/DeadNote.cs]: ProefExamen namespaces need to be under every other namspace

File: Assets/Scripts/Framework/Gameplay/LaneSystem/Lane.cs [hunk 5]
```diff
@@ -28,6 +28,9 @@ public class Lane : MonoBehaviour
         [SerializeField]
         private Vector3 _targetNotePosition;
 
+        [SerializeField]
+        private GameObject notePrefab;
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/Lane.cs]:
```suggestion
        private GameObject _notePrefab;
```

File: Assets/Scripts/Framework/Gameplay/LaneSystem/Note.cs [hunk 6]
```diff
@@ -40,6 +53,15 @@ public class Note : MonoBehaviour
         /// </summary>
         public float LerpAlpha => _lerpAlpha;
 
+        private void Start()
+        {
+            if(_animator == null)
+                _animator = gameObject.GetComponent<Animator>();
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/Note.cs:59]:
```cs
           if(_animator == null)
                _animator = gameObject.GetComponent<Animator>();

            if (_spriteRenderer == null)
                _spriteRenderer = gameObject.GetComponent<SpriteRenderer>();
```

GetComponent in `Awake`

File: Assets/Scripts/Framework/Gameplay/LaneSystem/Note.cs [hunk 7]
```diff
@@ -60,10 +82,24 @@ public void SetNoteValues(Lane lane,Vector2 initialPosition, Vector2 targetPosit
             transform.position = _initialPosition;
         }
 
+        public void HitNote()
```
Comment (PiterGroot) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/Note.cs:88]: Missing summery

**Review by PiterGroot — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Framework/Gameplay/LaneSystem/Note.cs [hunk 8]
```diff
@@ -60,10 +82,27 @@ public void SetNoteValues(Lane lane,Vector2 initialPosition, Vector2 targetPosit
...
+
         private void Update()
         {
             if (SessionValues.Instance.paused || _laneID == -1)
+            {
+                _animator.speed = 0;
                 return;
+            }
+            else
+                _animator.speed = _animationSpeed / SessionValues.Instance.travelTime;
```
Comment (lucasvdiepen) [inline Assets/Scripts/Framework/Gameplay/LaneSystem/Note.cs]: Remove unnecessary else statement
```suggestion

            _animator.speed = _animationSpeed / SessionValues.Instance.travelTime;
```

**Review by lucasvdiepen — CHANGES_REQUESTED** [review]
Looks good! Could you please remove all the underscores in the file names except for the prefixes and suffixes. Also remove the 512x512 in file names.

**Review by lucasvdiepen — APPROVED** [review]

**Pull request merged**